tests/test_compilesim.py

```diff
@@ -977,6 +1013,19 @@ def test_rom_out_of_range_error(self):
         with self.assertRaises(pyrtl.PyrtlError) as ex:
             self.sim(tracer=sim_trace)
 
+    def test_rom_romblock_in_memory_value_map_error(self):
+        rom_data_array = [1, 2, 3, 4, 5, 6, 7, 8]
+        rom1 = pyrtl.RomBlock(bitwidth=4, addrwidth=3, romdata=rom_data_array)
+        rom_add = pyrtl.Input(3, "rom_in_2")
+        rom_out = pyrtl.Output(4, "rom_out_2")
+        rom_out <<= rom1[rom_add]
+        mem_val_map = {rom1: {0: 0, 1: 1, 2: 2, 3: 3}}
+
+        sim_trace = pyrtl.SimulationTrace()
+        with self.assertRaises(pyrtl.PyrtlError) as error:
+            self.sim(tracer=sim_trace, memory_value_map=mem_val_map)
+        self.assertEqual(str(error.exception), "RomBlock in memory_value_map")
```

Comment:
Avoid checking for exact error messages in tests because it tightly couples the error code and test code - a minor spelling/grammar change to the error message will break this test, which makes it more difficult to modify PyRTL. I think the `assertRaises` is enough of a test, but if you want to be more specific maybe just check that the error message mentions `memory_value_map` somewhere?